### DISCUSSION TITLE
New module to watch Consul KV changes and update the context + fire RefreshEvent

### DISCUSSION THREAD

**Discussion by FrogDevelopper** [micronaut-projects/micronaut-discovery-client#676] [Ideas] [answered]
@graemerocher @sdelamo and other 
Hello,
I've created my own [micronaut-consul-watcher project](https://github.com/FrogDevelopment/micronaut-consul-watcher), and would like to know if you would be interested to have it as a real Micronaut module ?

If yes, it will of course be recreated following [New-Module-Checklist](https://github.com/micronaut-projects/micronaut-core/wiki/New-Module-Checklist).

**Comment by graemerocher** [comment]
Sure would make sense as part of https://github.com/micronaut-projects/micronaut-discovery-client/

**Reply by FrogDevelopper** [reply]
like as a `discovery-client/src/main/java/io/micronaut/discovery/consul/watch` package ?
I'll try (I'm not familiar with the integration tests using Spock & Groovy)

**Reply by sdelamo** [reply]
@FrogDevelopper you can write tests in Java and JUnit 5 as well. Let me know if you need help setting it up.

**Reply by FrogDevelopper** [reply]
@sdelamo 
Thanks
I'm currently writing Unit Tests in Groovy + Spock (make me learn a new test framework)
But, yes for Integration Tests I may have to use "classical" JUnit5
I'll cry for help if needed :)

**Reply by FrogDevelopper** [reply]
@sdelamo 
I've made a [1st try (based on 5.0.x)](https://github.com/micronaut-projects/micronaut-discovery-client/compare/5.0.x...FrogDevelopment:micronaut-discovery-client:feature/consul-watch)
I've created a new module `test-suite-consul-watch` that contains Integration testing based on different types of configuration (JSON, YAML, PROPERTIES or NATIVES)
Is it correct to do it like this ?

**Comment by FrogDevelopper** [comment]
@graemerocher , @sdelamo 
Which branch do your prefer I use as base ? 4.6.x or 5.0.x ?

**Reply by sdelamo** [reply, marked as answer]
@FrogDevelopper 4.6.x

**Comment by FrogDevelopper** [comment]
[PR](https://github.com/micronaut-projects/micronaut-discovery-client/pull/678) ready :)
